ionoscloud/resource_server.go

```diff
@@ -917,60 +930,59 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
-		_, apiResponse, err := client.ServersApi.DatacentersServersVolumesFindById(ctx, dcId, d.Id(), bootVolume).Execute()
-		logApiRequestTime(apiResponse)
-
-		if err != nil {
-			volume := ionoscloud.Volume{
-				Id: &bootVolume,
-			}
-			_, apiResponse, err := client.ServersApi.DatacentersServersVolumesPost(ctx, dcId, d.Id()).Volume(volume).Execute()
-			logApiRequestTime(apiResponse)
-			if err != nil {
-				diags := diag.FromErr(fmt.Errorf("an error occured while attaching a volume dcId: %s server_id: %s ID: %s Response: %s", dcId, d.Id(), bootVolume, err))
-				return diags
-			}
-
-			// Wait, catching any errors
-			_, errState = cloudapi.GetStateChangeConf(meta, d, apiResponse.Header.Get("Location"), schema.TimeoutCreate).WaitForStateContext(ctx)
-			if errState != nil {
-				diags := diag.FromErr(fmt.Errorf("an error occured while waiting for a state change for dcId: %s server_id: %s ID: %s %w", dcId, d.Id(), bootVolume, err))
-				return diags
-			}
-		}
 
 		properties := ionoscloud.VolumeProperties{}
+		inlineVolumeIds := d.Get("inline_volume_ids")
+
+		if inlineVolumeIds != nil {
+			inlineVolumeIds := inlineVolumeIds.([]interface{})
+			for i, volumeId := range inlineVolumeIds {
+				volumeIdStr := volumeId.(string)
+				volumePath := fmt.Sprintf("volume.%d.", i)
+				_, apiResponse, err := client.ServersApi.DatacentersServersVolumesFindById(ctx, dcId, d.Id(), volumeIdStr).Execute()
+				logApiRequestTime(apiResponse)
+				if err != nil {
+					diags := diag.FromErr(fmt.Errorf("an error occured while getting a volume dcId: %s server_id: %s ID: %s Response: %s", dcId, d.Id(), volumeId, err))
+					return diags
+				}
+				if v, ok := d.GetOk(volumePath + "name"); ok {
+					vStr := v.(string)
+					properties.Name = &vStr
+				}
 
-		if v, ok := d.GetOk("volume.0.name"); ok {
-			vStr := v.(string)
-			properties.Name = &vStr
-		}
+				if v, ok := d.GetOk(volumePath + "size"); ok {
+					vInt := float32(v.(int))
+					properties.Size = &vInt
+				}
 
-		if v, ok := d.GetOk("volume.0.size"); ok {
-			vInt := float32(v.(int))
-			properties.Size = &vInt
-		}
+				if v, ok := d.GetOk(volumePath + "bus"); ok {
+					vStr := v.(string)
+					properties.Bus = &vStr
+				}
 
-		if v, ok := d.GetOk("volume.0.bus"); ok {
-			vStr := v.(string)
-			properties.Bus = &vStr
-		}
+				if v, ok := d.GetOk(volumePath + "is_boot_volume"); ok {
+					vBool := v.(bool)
+					if vBool {
+						properties.SetBootOrder(constant.BootOrderPrimary)
+					}
+				}
 
-		_, apiResponse, err = client.VolumesApi.DatacentersVolumesPatch(ctx, d.Get("datacenter_id").(string), bootVolume).Volume(properties).Execute()
-		logApiRequestTime(apiResponse)
+				_, apiResponse, err = client.VolumesApi.DatacentersVolumesPatch(ctx, d.Get("datacenter_id").(string), volumeIdStr).Volume(properties).Execute()
+				logApiRequestTime(apiResponse)
+				if err != nil {
+					diags := diag.FromErr(fmt.Errorf("error patching volume (%s) (%w)", d.Id(), err))
+					return diags
```

Comment:
@cbeti-ionos same thing related to this `return` statement, do as you consider

Reply:
the fewer lines of code, the better - changed as wekk